File: cifrado/Escritorio/Python/Cifrado cesar/Cifrador.py
import TextoPlano
import Alfabeto

class Cifrador():
		
	def cifrarTipoCesar(texto, desp):
		alfabeto="aábcdeéfghiíjklmnñoópqrstuúvwxyzAÁBCDEÉFGHIÍJKLMNÑOÓPQRSTUÚVWXYZ";
		alfa = Alfabeto();
		textoCifrado = "";
		letra = "";
		posicion=0;
		for x in texto:
			print(textoCifrado);
			if alfa.buscar(x):
				posicion = alfa.buscarPosicionDeLetra(x);
				posicion = posicion + desp;
				if posicion > len(alfabeto):
					posicion = posicion - len(alfabeto);
					
				textoCifrado = textoCifrado + alfabeto[posicion];
			
			else:
				textoCifrado = textoCifrado + x;

		return textoCifrado;


	def cifrarTipoTransposicionInversa(texto):
		text = TextoPlano();
		return text.invertirTexto(texto);


	def cifrarTipoTransposicionInversaGrupos(texto,desplazamiento):
		text = TextoPlano();
		return text.invertirEnGrupos(texto,desplazamiento);

File: cifrado/Escritorio/codigo/Ruby/Cifrador.rb
requiere 'Alfabeto'
requiere 'TextoPlano'

class Cifrador

	def initialize(algo)
		@iii=algo

	def cifrar(texto, desp)
		textoCifrado = ""
		letra = ""
		posicion=0
		alfabeto="aábcdeéfghiíjklmnñoópqrstuúvwxyzAÁBCDEÉFGHIÍJKLMNÑOÓPQRSTUÚVWXYZ"
		 texto.each_char do |caracter|

   			if buscar(caracter)
				posicion = buscarPosicionDeLetra(caracter)
				posicion = posicion + desp
				if posicion > alfabeto.length
					posicion = posicion - alfabeto.length
				end

				textoCifrado = textoCifrado + alfabeto[posicion]
			
			else
				textoCifrado = textoCifrado + caracter
			end

		end 


		return textoCifrado
	end



	def cifrarTipoTransposicion(texto)
		tp = TextoPlano.new()
		return tp.invertirTexto(texto)


	def cifrarTipoTransposicionGrupos(texto,desplazamiento)
		tp = TextoPlano.new()
		return tp.invertirEnGrupos(texto,desplazamiento)